server/neptune/workflows/internal/activities/github.go

```diff
@@ -204,3 +213,33 @@ func (a *githubActivities) FetchRoot(ctx context.Context, request FetchRootReque
 		LocalRoot: localRoot,
 	}, nil
 }
+
+type CompareCommitRequest struct {
+	DeployRequestRevision  string
+	LatestDeployedRevision string
+	Repo                   internal.Repo
+}
+
+type CompareCommitResponse struct {
+	CommitComparison DiffDirection
+}
+
+func (a *githubActivities) CompareCommit(ctx context.Context, request CompareCommitRequest) (CompareCommitResponse, error) {
+	client, err := a.ClientCreator.NewInstallationClient(request.Repo.Credentials.InstallationToken)
+	if err != nil {
+		return CompareCommitResponse{}, errors.Wrap(err, "creating installation client")
+	}
+
+	comparison, resp, err := client.Repositories.CompareCommits(ctx, request.Repo.Owner, request.Repo.Name, request.LatestDeployedRevision, request.DeployRequestRevision, &github.ListOptions{})
+	if err != nil {
+		return CompareCommitResponse{}, errors.Wrap(err, "comparing commits")
+	}
+
+	if comparison.GetStatus() == "" || resp.StatusCode != http.StatusOK {
+		return CompareCommitResponse{}, errors.New("invalid commit comparison status")
```

Comment:
If i saw this error i wouldn't know what to do with it. Can you at least add the reason in there? Like status code and/or empty status?  (Is empty status even possible)?